lib/OpenQA/WebSockets.pm

```diff
@@ -14,82 +14,126 @@
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
 package OpenQA::WebSockets;
+use Mojo::Base 'Mojolicious';
 
-use strict;
-use warnings;
-
-use base 'Net::DBus::Object';
-
-use Net::DBus::Exporter 'org.opensuse.openqa.WebSockets';
-use Mojo::IOLoop;
+use Mojo::Server::Daemon;
 use OpenQA::IPC;
-use OpenQA::WebSockets::Server;
-use OpenQA::Utils 'log_debug';
-
-# monkey patching for debugging IPC
-sub _is_method_allowed {
-    my ($self, $method) = @_;
-
-    my $ret = $self->SUPER::_is_method_allowed($method);
-    if ($ret) {
-        log_debug "IPC calling $method";
-    }
-    return $ret;
+use OpenQA::Setup;
+use OpenQA::Utils qw(log_debug log_warning log_info);
+use OpenQA::WebSockets::Model::Status;
+use db_profiler;
+
+sub run { __PACKAGE__->new->setup->run }
+
+sub startup {
+    my $self = shift;
+
+    $self->defaults(appname => 'openQA Websocket Server');
+    $self->mode('production');
+
+    # no cookies for worker, no secrets to protect
+    $self->secrets(['nosecretshere']);
+    $self->config->{no_localhost_auth} ||= 1;
+
+    # Assetpack is required to render layouts pages
```

Comment:
I'm aware that you've only moved this line but I'm wondering why the web socket server would render layout pages.

Reply:
Yea, i've been trying very hard to avoid any logic changes during the refactoring. But AssetPack is indeed a pretty heavy dependency that seems useless here. I'll test removing it.